### PR TITLE
Templatize the certification pipeline

### PR DESCRIPTION
**Pull request by ashuka24**
This PR refactors the certification pipeline to use the newly introduced macros and sub-templates introduced by https://github.com/greenplum-db/pxf/pull/1053. 

It introduces new sub-templates for the following jobs:
- certify gpdb rc candidate against latest pxf release 
- certification reporting gate

With this refactor, the `certification_pipeline.yml` was renamed tand moved to `templates/certification_pipeline-tpl.yml`.

This PR also does some renaming: 
- underscores were changed to dashes
- removed some extraneous dashes
- removed `on` from job names
- changed the platform name to be consistent with other pipelines
Ex:
  `certify_gpdb-5_with_pxf-gp5_on_rhel7` -> `certify-gpdb5-with-pxf-gp5-centos7`

### CODE REVIEW

File: concourse/pipelines/templates/jobs/certify-gpdb-with-pxf-tpl.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    <<: *slack_alert
+{% endif %}
+    file: pxf_src/concourse/tasks/test_certification.yml
+    image: gpdb[[x.gp_ver]]-pxf-dev-[[x.test_platform]]-image
+    params:
+      ACCESS_KEY_ID: ((tf-machine-access-key-id))
+      GP_VER: [[x.gp_ver]]
+      GROUP: gpdb,proxy,profile
+      SECRET_ACCESS_KEY: ((tf-machine-secret-access-key))
+  - task: upload_certification_for_gpdb-[[x.gp_ver]]_with_pxf-gp[[x.gp_ver]]-hdp2_on_[[x.test_platform]]
```
Comment (yimingli-vmware) [inline concourse/pipelines/templates/jobs/certify-gpdb-with-pxf-tpl.yml:36]: for the `task` on line 25, we switched `_` to `-`, do we want to do the same thing here?

File: concourse/pipelines/templates/jobs/certification-reporting-gate-tpl.yml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  plan:
+  - in_parallel:
+    - get: pxf_src
+    # gpdb release candidate tarballs and PXF packaqges used in testing jobs
+{% for gpdb_tarball_name in job_names.gpdb_tarballs.keys() | list %}
+    {# only add the job to the list if the job_name contains the string `gpdb[[x.gp_ver]]` #}
+    {% if 'gpdb' ~ x.gp_ver in gpdb_tarball_name %}
+    - get: [[gpdb_tarball_name]]
+      passed:
+        - [[job_names.gpdb_tarballs[gpdb_tarball_name]]]
```
Comment (yimingli-vmware) [inline concourse/pipelines/templates/jobs/certification-reporting-gate-tpl.yml:19]: this is a key value pair, here the value of this given `gpdb_tarball_name` is the `certify` job name, right?

File: concourse/pipelines/templates/jobs/certify-gpdb-with-pxf-tpl.yml [hunk 3]
```diff
@@ -0,0 +1,42 @@
+{# Template for the PXF certification job definition.
+ #
+ # Expects a dictionary with the name 'x' to be set as defined in macros.j2 file
+ #
+ #}
+{% do x.update({'job_name': 'certify-gpdb' ~ x.gp_ver ~ '-with-pxf-gp' ~ x.gp_ver ~ '-' ~ x.test_platform}) %}
```
[the remark this answers is not in the document]
Comment (ashuka24) [reply concourse/pipelines/templates/jobs/certify-gpdb-with-pxf-tpl.yml:6]: Correct, as part of this refactoring, underscores were changed to dashes, we removed `on`, and made platform names consistent with those present in the build and dev pipelines. 

I have added a section explaining this change in the PR description

File: concourse/pipelines/templates/certification_pipeline-tpl.yml [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+
+## ======================================================================
+## RESOURCES
+## ======================================================================
+resources:
+    [[macros.resource_pxf_dependencies('automation')]]
+    [[macros.resource_singlecluster('hdp2')]]
+
+## ---------- Github Repos ----------
+- name: pxf_src
```
[the remark this answers is not in the document]
Comment (ashuka24) [reply concourse/pipelines/templates/certification_pipeline-tpl.yml:47]: this pxf_src is sepcial because we look at a tag filter instead of files. i did not want to over-complicate the existing github repos macro but did not feel like a separate macro/resource subtemplate was needed here as this is the only place we would use it.

File: concourse/pipelines/templates/jobs/certification-reporting-gate-tpl.yml (hunk 2 above)
Comment (ashuka24) [reply concourse/pipelines/templates/jobs/certification-reporting-gate-tpl.yml:19]: correct, the value here will be the certify job that the `gpdb_tarball_name` passed

File: concourse/pipelines/templates/resources/pxf-released-artifact-tpl.yml [hunk 5]
```diff
@@ -0,0 +1,16 @@
+{# Template for a PXF build artifact resource definition.
+ #
+ # Expects a dictionary with the name 'x' to be set as defined in macros.j2 file
+ #}
+
+{# include PXF artifact only if both build and test platform are the same to avoid duplicates
+ # for example, the 'oel7' test platform does not have an 'oel7' build job and uses 'centos7' artifacts
+ #}
```
Comment (denalex) [inline concourse/pipelines/templates/resources/pxf-released-artifact-tpl.yml:8]: obsolete comment ?

**Review by denalex — APPROVED** [review]

File: concourse/pipelines/templates/certification_pipeline-tpl.yml [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+{% call(x) macros.for_each_config(build_test_pxf_combinations) %}
+    {% include 'resources/pxf-test-image-tpl.yml' %}
+{% endcall %}
+
+    [[macros.resource_registry_image('ccp-7-image', 'gcr.io/data-gpdb-public-images/ccp', 'latest')]]
+
+## ---------- Greenplum Release Candidate Tarballs ----------
+{% call(x) macros.for_each_config(build_test_pxf_combinations) %}
+    {% if x.gp_ver == '5' %}
+        {% include 'resources/greenplum-5-release-candidate-tarball-tpl.yml' %}
```
[the remark this answers is not in the document]
Comment (ashuka24) [reply concourse/pipelines/templates/certification_pipeline-tpl.yml:65]: not quite. the gpdb5 tarball comes from aws. the gpdb6 and 7 tarballs come from gcp and so the base resource definition is quite different. removing the if block here would just create one in the resource sub-template.

**Review by yimingli-vmware — APPROVED** [review]
